src/RankedPairsCalculator.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace PivotLibre\Tideman;
+
+use \InvalidArgumentException;
+use \Exception;
+
+use PivotLibre\Tideman\Agenda;
+use PivotLibre\Tideman\Candidate;
+use PivotLibre\Tideman\MarginList;
+use PivotLibre\Tideman\CandidateSet;
+use PivotLibre\Tideman\CandidateList;
+use PivotLibre\Tideman\RankedPairsGraph;
+use PivotLibre\Tideman\ListOfMarginLists;
+use PivotLibre\Tideman\CandidateComparator;
+use PivotLibre\Tideman\TieBreakingMarginComparator;
+
+class RankedPairsCalculator
+{
+    private $tieBreakingMarginComparator;
+    private $tieBreakingCandidateComparator;
+    /**
+     * Constructs a Ranked Pairs Calculator, verifying that the specified tie-breaking ballot contains no ties.
+     * Retains a copy of the tie-breaking Ballot so that the caller may modify the parameterized Ballot without
+     * affecting this class.
+     * @param tieBreakingBallot
+     */
+    public function __construct(Ballot $tieBreakingBallot)
+    {
+        if ($tieBreakingBallot->containsTies()) {
+            throw new InvalidArgumentException("Tie breaking ballot must not contain any ties. $tieBreakingBallot");
+        } else {
+            $myTieBreakingBallot = clone $tieBreakingBallot;
+            $tieBreaker = new TotallyOrderedBallotMarginTieBreaker(new CandidateComparator($myTieBreakingBallot));
+            $this->tieBreakingMarginComparator = new TieBreakingMarginComparator($tieBreaker);
+            $this->tieBreakingCandidateComparator = new CandidateComparator($myTieBreakingBallot);
+        }
+    }
+
+    /**
+     * @param int number of winners to return.
+     * @param NBallot the ballots from the electorate.
+     * @return CandidateList in which the zeroth Candidate is the most preferred, the first Candidate is the second most
+     * preferred, and so on until the last Candidate who is the least preferred. Ties are broken according to the Ballot
+     * provided to the constructor.
+     */
+    public function calculate(int $numWinners, NBallot ...$nBallots) : CandidateList
```

Comment:
This is best place to start reviewing. It is also most probably the method that the Laravel app will use to get the winner. Do you think this method is workable from the Laravel app's perspective?